### ISSUE TITLE
warning: variable does not need to be mutable

### ISSUE THREAD

**Issue by compressed** [colin-kiegel/rust-derive-builder#104]
Hi,
I'm receiving an `unused_mut` lint warning with the following:

```rust
#[macro_use]
extern crate derive_builder;

#[derive(Builder)]
struct A {
    pub name: String,
}
```

Gives:

```
[Running 'cargo check']
   Compiling derive_test v0.1.0 (file:///home/chris/code/derive_test)
warning: variable does not need to be mutable
 --> src/lib.rs:5:10
  |
5 | #[derive(Builder)]
  |          ^^^^^^^
  |
  = note: #[warn(unused_mut)] on by default

    Finished dev [unoptimized + debuginfo] target(s) in 0.11 secs
[Finished running]

```

**Comment by colin-kiegel** [comment]
Thanks for the report. :smile: :+1:

I probably fixed it already in git. But CI is currently complaining about something else, which blocks a new release.

**Comment by compressed** [comment]
Oh cool, I did not think to check the git version. You're right, the warning is fixed there. Thanks!

**Comment by Enet4** [comment]
When is there going to be a new release? It's been a while, and I believe more people would like not to receive these warnings.

**Comment by colin-kiegel** [comment]
Right, I'll try to look into it this weekend.

**Comment by Fiedzia** [comment]
Being displayed once for each field, it really pollutes warnings, can we have a release with this fix?

**Comment by softprops** [comment]
this is happening to me as well

**Comment by fluffysquirrels** [comment]
As a workaround, you can silence the warning in your module with an `#!allow(unused_mut)]` directive. Note that adding `#[allow(unused_mut)]` to the struct defintion doesn't seem to work. I've restricted the scope of this directive in an inline sub-module so it only silences this instance of the warning, like so:

```rust
// In ${crate}/lib.rs
mod http_driver {
    // Allow lint `unused_mut`, which occurs in the HttpDriver's derived Builder
    // implementation.
    // This inline sub-module is only here to minimise the scope of this allow.
    // Issue: https://github.com/colin-kiegel/rust-derive-builder/issues/104
    #![allow(unused_mut)]

    /// A driver using a pre-existing WebDriver HTTP URL.
    #[derive(Builder)]
    #[builder(field(private))]
    pub struct HttpDriver {
        #[builder(setter(into))]
        pub url: String,
    }
}

pub use self::http_driver::{HttpDriver, HttpDriverBuilder};
```

Hope that helps.

**Comment by fluffysquirrels** [comment]
Another workaround that's a bit quicker to apply is to add a `[patch]` section to the Cargo.toml of your crate or workspace to use the unreleased version of `derive_builder`, like so:

```toml
# Use a pre-release version of derive_builder that fixes issue
# <https://github.com/colin-kiegel/rust-derive-builder/issues/104>
[patch.crates-io.derive_builder]
git = "https://github.com/colin-kiegel/rust-derive-builder"
rev = "2585c5b1cbfacfc3450deafb0e3df6055da38bf7"
```

**Comment by fluffysquirrels** [comment]
This is now fixed in [tag v0.5.1](https://github.com/colin-kiegel/rust-derive-builder/releases/tag/v0.5.1) and the fix is [published to crates.io](https://crates.io/crates/derive_builder/0.5.1).

Ready to close, I think.
